### PR TITLE
Get rid of preprocessor conditions in Systray::calcTrayIconCenter()

### PR DESCRIPTION
**Pull request by sbeyer**

### CODE REVIEW

**Comment by sbeyer** [comment]
I really need some VMs with Unity, Gnome, XFCE, LXDE, i3, etc. to test, but under KDE it works fine. But I don't have that... so can somebody test it in Gnome at least?

I would really be surprised if some garbage `QRect` other than the default-constructed one is returned.

File: src/gui/systray.cpp [hunk 1]
```diff
@@ -360,15 +360,16 @@ QPoint Systray::computeWindowPosition(int width, int height) const
...
-#if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
-    auto trayIconCenter = geometry().center();
-    return trayIconCenter;
-#else
-    // On Linux, fall back to mouse position (assuming tray icon is activated by mouse click)
-    return QCursor::pos();
-#endif
+    QRect iconRect = geometry();
+
+    if (iconRect == QRect()) {
```
Comment (er-vin) [inline src/gui/systray.cpp:365]: `iconRect.isNull()` would be more idiomatic (and slightly faster IIRC).

File: src/gui/systray.cpp [hunk 2]
```diff
@@ -360,15 +360,15 @@ QPoint Systray::computeWindowPosition(int width, int height) const
...
-    // QSystemTrayIcon::geometry() is broken for ages on most Linux DEs (invalid geometry returned)
-    // thus we can use this only for Windows and macOS
-#if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
-    auto trayIconCenter = geometry().center();
-    return trayIconCenter;
-#else
-    // On Linux, fall back to mouse position (assuming tray icon is activated by mouse click)
-    return QCursor::pos();
-#endif
+    QRect iconRect = geometry();
```
Comment (er-vin) [inline src/gui/systray.cpp:363]: bah stupid me, only realize it now... minor nitpick: `const auto iconRect`?

Comment (sbeyer) [reply src/gui/systray.cpp:363]: Why stupid? I think it's quite natural in reviewing to find some issues later.

**Comment by sbeyer** [comment]
Sorry, for the re-request of the review. I wanted to see what that functionality actually does.

**Comment by mgallien** [comment]
@sbeyer Do you still plan to work on this or should we just close it ?
As a first step, could you rebase on latest mater ?

**Comment by sbeyer** [comment]
> @sbeyer Do you still plan to work on this or should we just close it?

I am sorry, I don't have time to work on nextcloud any longer :( I'd say I close this PR then.

**Pull request closed without merging**